### ISSUE TITLE
Size restriction in Viewer??

### ISSUE THREAD

**Issue by G-Thomson** [UoA-eResearch/easyD3R#1]
Hi Nick

Sorry I've been busy the last couple of weeks to do much on this project however I have had a quick go with it this afternoon and ran into an issue. The sequence I was trying to draw is 2000bases long. However this seems greater than the viewer allows so I can't visualise the full sequence. I had a quick skim through the R and js code and couldn't see anything like a hard limit I could increase although the output was set to `width = '100%'  `

I also had a quick go creating the beginnings of wrapper functions for each element hopefully creating a more user friendly input method than the lists of lists.

Here is my example code:
```
library("easyD3R")

# Base instance of sequence
geneVR <- function(length, height = 10, fill = "gray"){
  inst <- list(list(type = "rect", x = 0, y = 80, width = length, height = height, fill = fill))
  easyD3(inst)
}

# Rectangular features
feat_rect <- function(coords, height = 20, fill = "orange"){
  inst <- purrr::flatten(purrr::by_row(fta1, ..f = function(x) 
    rec <- list(type = "rect", 
                 x = as.numeric(x[1]), 
                 y = 75, 
                 width = as.numeric(x[2]-x[1]),
                 height = height, 
                 fill = fill), 
    .labels = FALSE))
  return(inst)
}

# Example data
fta1 <- structure(list(start = c(126, 433, 603, 1604), 
                       end = c(327, 495, 644, 1831)), 
                  .Names = c("start", "end"), 
                  row.names = c(NA, -4L), 
                  class = "data.frame")

# Implement visualisation
geneVR(2000) + 
  feat_rect(fta1, fill = "blue")
``` 

See you at Razbaz.

Cheers

Geoffrey

**Comment by neon-ninja** [comment]
Hmm, executing your above code generates this image for me ![image](https://cloud.githubusercontent.com/assets/3378822/23531182/1d2028de-000a-11e7-8241-327e79e6c114.png)

So seems to work fine on my machine. Can you clarify the exact steps you took to reproduce this bug? What are you viewing it in? RStudio or chrome? If Rstudio, what version? What's the resolution of the screen you're trying to view it on?

**Comment by G-Thomson** [comment]
That is correct, I got the same image. However the grey rectangle is supposed to be 2000 units long with four blue rectangles on it, not three. See the example data.

Currently the image is only about 1100 wide but some genes I want to draw are ~20,000 units wide.

I am viewing it in both RStudio (0.99.903) and Firefox (50.1.0).

**Comment by neon-ninja** [comment]
Ah, I see. The function easyD3 takes width as an argument, which is the width of the entire plot (defaults to window width). If you set that to the width of the grey rectangle it should work. So changing `easyD3(inst)` to `easyD3(inst, width = length)` generates 
![image](https://cloud.githubusercontent.com/assets/3378822/23531999/14490ae2-000e-11e7-84b0-5e7d19ca8f25.png)

Perhaps the library could be made a bit smarter, by checking the width of the passed initialising objects and setting the width of the canvas to that...

**Comment by neon-ninja** [comment]
Hmm, seems RStudio won't allow me to export an image wider than 1820 pixels..

**Comment by neon-ninja** [comment]
An image 20K pixels wide may not be particularly reasonable anyway, we may want to do some automatic scaling to generate images that are all the same width regardless of the number of bases

**Comment by G-Thomson** [comment]
I agree, that sounds sensible. Do you think it would be best to scale the data before putting it into easyD3 or after we have put it in?

Maybe we should just scale the introns (regions between blue rectangles) so that relative distances between the blue rectangles are retained and the interesting features are easy to see?

**Comment by neon-ninja** [comment]
I think a uniform scale within https://github.com/UoA-eResearch/easy-d3/blob/master/easy-d3.js would make sense. To make it easier for users of the library, to not have to repeat the same boilerplate code. We could even do some feature viewer style interactive zooming

**Comment by neon-ninja** [comment]
Here's what the above example looks like using this scaling code - https://github.com/UoA-eResearch/easyD3R/commit/5d99cf85cd408fa31f5db73a891e486f199e913d

![image](https://cloud.githubusercontent.com/assets/3378822/23538047/2d345b66-0035-11e7-8ed1-486fecf8626a.png)
